protostar/argument_parser/abi_obtainer.py

```diff
@@ -0,0 +1,17 @@
+from typing import Any
+
+from protostar.cli import NetworkCommandUtil
+from protostar.starknet_gateway import (
+    AbiResolver,
+    DataTransformerPolicy,
+)
+
+
+async def abi_from_args(args: Any):
+    network_command_util = NetworkCommandUtil(args)
+    gateway_client = network_command_util.get_gateway_client()
+    abi_resolver = AbiResolver(client=gateway_client)
+    data_transformer_policy = DataTransformerPolicy(abi_resolver=abi_resolver)
+    return await data_transformer_policy.resolve_abi_or_fail(
+        address=args.contract_address
+    )
```

Comment:
`ArgumentParser` component should be unaware of Protostar. It should be actually moved out of the Protostar directory. `protostar.cli` would be a better place for this function.